### ISSUE TITLE
Failure to Build/Install on Windows

### ISSUE THREAD

**Issue by Teh-Bobo** [cargo-checkmate/cargo-checkmate#34]
When trying to run "cargo install cargo-checkmate" on my Windows machine it fails with the following errors:

(username masked)
> error[E0433]: failed to resolve: could not find `unix` in `os`
>   --> C:\Users\\***\\.cargo\registry\src\github.com-1ecc6299db9ec823\cargo-checkmate-0.1.10\src\srcbundle.rs:78:18
>    |
> 78 |     use std::os::unix::fs::PermissionsExt;
>    |                  ^^^^ could not find `unix` in `os`
> 
> error[E0599]: no method named `set_mode` found for struct `Permissions` in the current scope
>   --> C:\Users\\***\\.cargo\registry\src\github.com-1ecc6299db9ec823\cargo-checkmate-0.1.10\src\srcbundle.rs:82:11
>    |
> 82 |     perms.set_mode(perms.mode() | 0o500);
>    |           ^^^^^^^^ method not found in `Permissions`
> 
> error[E0599]: no method named `mode` found for struct `Permissions` in the current scope
>   --> C:\Users\\***\\.cargo\registry\src\github.com-1ecc6299db9ec823\cargo-checkmate-0.1.10\src\srcbundle.rs:82:26
>    |
> 82 |     perms.set_mode(perms.mode() | 0o500);
>    |                          ^^^^ method not found in `Permissions`
> 
> Some errors have detailed explanations: E0433, E0599.
> For more information about an error, try `rustc --explain E0433`.
> error: failed to compile `cargo-checkmate v0.1.10`, intermediate artifacts can be found at `C:\Users\***\AppData\Local\Temp\cargo-installZpXgHC`
> 
> Caused by:
>   could not compile `cargo-checkmate` due to 3 previous errors

The errors seem to be cause by this unix specific function:
```
fn make_executable(f: std::fs::File) -> IOResult<()> {
    use std::os::unix::fs::PermissionsExt;

    let mut perms = f.metadata()?.permissions();
    // Set user read/write perms on unix:
    perms.set_mode(perms.mode() | 0o500);
    f.set_permissions(perms)?;
    Ok(())
}
```

I didn't dig super into the details of the call path for it, but maybe this needs to gated by OS with a Windows version as well? That way this function is only attempted to be built on unix-like systems.

**Comment by Teh-Bobo** [comment]
I should also note I tried both versions 0.1.11 and 0.1.10

**Comment by nathan-at-least** [comment]
Yes, apologies, that function is unix specific, and the proper fix would be to have separate implementations of `make_executable` on each platform (or find a library that does this for all platforms). I'm not familiar with how to write the windows version, yet.

**Comment by nathan-at-least** [comment]
Ok, the only reason for changing a file to executable is for git hooks, so my next step is to figure out how git hooks work on windows and proceed from there.

**Comment by nathan-at-least** [comment]
I added the `help wanted` label because I do not have convenient access to a windows development system and would really appreciate adding windows support (especially with Github CI coverage).
